core/src/rpc.rs

```diff
@@ -60,6 +59,19 @@ fn new_response<T>(bank: &Bank, value: T) -> Result<RpcResponse<T>> {
     Ok(Response { context, value })
 }
 
+pub fn is_confirmed_rooted(
+    block_commitment_cache: &BlockCommitmentCache,
+    blockstore: &Blockstore,
+    slot: Slot,
+) -> bool {
+    slot <= block_commitment_cache.largest_confirmed_root()
+        && (blockstore.is_root(slot)
+            || block_commitment_cache
+                .bank()
+                .status_cache_ancestors()
+                .contains(&slot))
+}
```

Comment:
Having this logic in rpc seems a little weird to me, but there isn't another obvious place for it. 🤷‍♀️

Reply:
I agree, but it'll be easy to move one standalone function later.